<a href="https://colab.research.google.com/github/dhanushmajji1428/FlyRank/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/dhanushmajji1428/FlyRank/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My Lane

I selected the Content Refresh Recommendation lane because it focuses on identifying web pages that should be updated to improve search performance. This problem is useful because it helps prioritize content updates using data instead of manual decisions.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

 Research Question

Can machine learning identify which web pages should be refreshed to improve search performance?

Decision:
Decide which pages should be updated.

Action:
Refresh the selected pages.

Cost of a wrong recommendation:
Refreshing the wrong pages wastes time and resources, while missing important pages may reduce website traffic.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
!git clone https://github.com/dhanushmajji1428/FlyRank.git
%cd FlyRank

Cloning into 'FlyRank'...
remote: Enumerating objects: 119, done.
remote: Counting objects: 100% (119/119), done.
remote: Compressing objects: 100% (75/75), done.
remote: Total 119 (delta 35), reused 97 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (119/119), 1.84 MiB | 14.19 MiB/s, done.
Resolving deltas: 100% (35/35), done.
/content/FlyRank


In [4]:
!pwd
!ls


/content/FlyRank
AGENTS.md  DATA_USE.md	LICENSE    README.md	     SETUP.md	 work
CLAUDE.md  docs		notebooks  requirements.txt  skills
data	   GUIDE.md	outputs    scripts	     submission


In [5]:
import os

for root, dirs, files in os.walk("data"):
    for file in files:
        print(os.path.join(root, file))

data/raw/content_refresh_anonymized.csv


In [6]:
import pandas as pd

# Load the dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Show 3 real numbers
print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("\nColumn names:")
print(df.columns.tolist())

print("\nMissing values:")
print(df.isnull().sum().sum())

Number of rows: 30000
Number of columns: 44

Column names:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']

Missing values:
73868


. Quick Look at the Data

The starter dataset contains 30,000 rows and 44 columns, providing a large amount of information about content performance. The dataset includes features such as search volume, impressions, clicks, sessions, CTR, average position, engagement rate, and trend direction. There are 73,868 missing values, which indicates that data preprocessing and cleaning will be important before building a machine learning model. These statistics show that the dataset is suitable for developing a content refresh recommendation system.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

 4. Careful Words: What I Can and Can't Claim

This project is intended to provide decision support by identifying patterns in historical content performance data. The results can help recommend which pages may benefit from a content refresh based on observed trends and measured metrics.

However, this project cannot prove causation or guarantee that refreshing a page will improve search rankings or traffic. Search performance depends on many external factors, including search engine algorithm updates, competition, and user behavior. Therefore, the model should be used as a recommendation tool rather than a guarantee of future performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.